xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -867,6 +890,36 @@ void CXBMCApp::onNewIntent(CJNIIntent intent)
   }
 }
 
+void CXBMCApp::onActivityResult(int requestCode, int resultCode, CJNIIntent resultData)
+{
+  for (auto it = m_activityResultEvents.begin(); it != m_activityResultEvents.end(); ++it)
+  {
+    if ((*it)->GetRequestCode() == requestCode)
+    {
+      m_activityResultEvents.erase(it);
+      (*it)->SetResultCode(resultCode);
+      (*it)->SetResultData(resultData);
+      (*it)->Set();
+      break;
+    }
+  }
+}
+
+int CXBMCApp::WaitForActivityResult(const CJNIIntent &intent, int requestCode, CJNIIntent &result)
+{
+  int ret = 0;
+  CActivityResultEvent* event = new CActivityResultEvent(requestCode);
```

Comment:
I should probably have said "manually managing memory using new and delete has been deprecated for 6 year" (because nobody is that good), I stand corrected;)